brian2/core/network.py

```diff
@@ -1035,21 +1035,17 @@ def after_run(self):
                 obj.after_run()
 
     def _nextclocks(self):
-        clocks_times_dt = [
-            (c, self._clock_variables[c][1][0], self._clock_variables[c][2][0])
-            for c in self._clocks
-        ]
-        minclock, min_time, minclock_dt = min(clocks_times_dt, key=lambda k: k[1])
+
+        minclock = min(self._clocks)
```

Comment:
While this (i.e. defining `__lt__` and `__le__`) is less of an issue than the equality check, for consistency/readability we should probably also have an explicit function here, or simply use something along the lines of `min(self._clocks, key=lambda c: c.t)`